src/metabase/analytics/stats.clj

```diff
@@ -600,19 +600,19 @@
   (u/prog1 eid-translation/default-counter
     (setting/set-value-of-type! :json :entity-id-translation-counter <>)))
 
-(defn- categorize-query-execution [{client :embedding_client executor :executor_id}]
+(defn- categorize-query-execution [{:keys [context embedding_client executor_id]}]
   (cond
-    (= "embedding-sdk-react" client)                     "sdk_embed"
-    (and (= "embedding-iframe" client) (some? executor)) "interactive_embed"
-    (and (= "embedding-iframe" client) (nil? executor))  "static_embed"
-    (and (#{"" nil} client) (nil? executor))             "public_link"
-    :else                                                "internal"))
+    (= "embedding-sdk-react" embedding_client)                        "sdk_embed"
+    (and (= "embedding-iframe" embedding_client) (some? executor_id)) "interactive_embed"
+    (and (= "embedding-iframe" embedding_client) (nil? executor_id))  "static_embed"
+    (some-> context name (str/starts-with? "public-"))                "public_link"
```

Comment:
I observed an executor id of 1 on a public view, so I don't think it needs to be in `#{"", nil}`.